Exit wptrunner with an error status if no tests ran

jgraham opened the pull request:
Typically no tests completing means that some error occured, or the user
supplied an invalid test path. In either case we should log the
problem and return a failure code.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/w3c/web-platform-tests/10030)
<!-- Reviewable:end -->

jgraham committed:
Exit wptrunner with an error status if no tests ran

Typically no tests completing means that some error occured, or the user
supplied an invalid test path. In either case we should log the
problem and return a failure code.

Ms2ger commented on tools/wptrunner/wptrunner/wptrunner.py:
```diff
@@ -267,14 +269,18 @@ def run_tests(config, test_paths, product, **kwargs):
-    return unexpected_total == 0
+    if test_total == 0:
+        logger.error("No tests ran")
+    return test_total > 0 and unexpected_total == 0
```
Nit:

```python
    if test_total == 0:
        logger.error("No tests ran")
        return False

    return unexpected_total == 0
```

Ms2ger commented on tools/wptrunner/wptrunner/testrunner.py:
```diff
@@ -789,5 +793,11 @@ def stop(self):
+    def test_count(self):
+        return sum(item.test_count for item in self.pool)
+
+    def subtest_count(self):
```
Unused?

gsnedders replied:
Yeah, this seems dead.

w3c-bots commented:
# Build BROKEN

Started: 2018-03-14 15:27:07
Finished: 2018-03-14 15:44:33


<h2>Failing Jobs</h2>
<ul>

<li>wptrunner_infrastructure</li>

</ul>




View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/25947)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/353378363)

jgraham commented:
Of course landing this is hard until the underlying  issue from #9979 is fixed.

gsnedders commented:
Should we just drop Chrome given #9979 so we can land this (and ensure Firefox doesn't regress), given that's not actually regressing testing Chrome? I'd be okay with that, and then #9979 becomes "re-enable Chrome" instead of "Chrome runs nothing".

jgraham committed:
Stop running Chrome wptrunner infrastructure tests for bustage

The tests were failing with some kind of Chromedriver/Chrome version
mismatch, so we can't block on these tests until that issue is fixed.

gsnedders commented on testharness_runner.html:
```diff
@@ -1,6 +0,0 @@
-<script>
-var timeout_multiplier = 1;
-var win = null;
-</script>
```
This seems unrelated to this commit.

jgraham commented:
OK, I didn't mind the subtest count existing even though it's not yet used.

Reverted the inadvertent change.
